### ISSUE TITLE
[Sub-Issue] [EVAL] Cálculo de Métricas Especializadas (Perplexidade, Acurácia, F1)

### ISSUE THREAD

**Issue by fabiooliveir** [fabiooliveir/train-model-mini#20]
### Descrição
Desenvolver funções de cálculo de métricas de qualidade para tarefas de Causal LM e classificação textual.

### Escopo Técnico
- Criar módulo `src/train_model_mini/evaluation/metrics.py`.
- Cálculo de **Perplexidade (PPL)** via `torch.exp(eval_loss)`.
- Suporte a métricas de classificação via biblioteca `evaluate` (Acurácia, F1-Score Macro/Weighted, Precisão, Recall).
- Serialização das métricas calculadas em arquivo JSON para versionamento de experimentos.

### Critérios de Aceitação
- [ ] Cálculo matemático correto de perplexidade a partir da cross-entropy loss.
- [ ] Relatório de métricas exportado em formato padronizado JSON.

---
- **Parent Epic:** #18
- **Blocked by:** #19
- **Blocks:** #25
